network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt

```diff
@@ -35,7 +38,15 @@ interface WgServerApi {
         val gateway: String,
     )
 
-    suspend fun registerPublicKey(publicKey: String): WgServerData?
+    suspend fun registerPublicKey(
+        publicKey: String,
+        mode: Mode = Default,
```

Comment:
I think `mode` should be nullable rather than forcing a value on it and then have a `Default` mode that is (and will never) be used.
```suggestion
        mode: Mode? = null,
```